### PR TITLE
WPS notifier retry

### PR DESCRIPTION
**Pull request by dnahodil**
@jkburges thought we might be able to use an Apache library to achieve the same result I'm happy to go down that path and will just leave it up to @jkburges.

Also, I'm thinking that these new parameters should not be passed into the constructor but maybe be passed into the execute method. But this will only be nice if you can configure a default so that they're not required all the time (possibly can be done with the DescribeParameter annotation).

### CODE REVIEW

File: src/extension/notifier/src/main/java/au/org/emii/wps/NotifierProcess.java [hunk 1]
```diff
@@ -39,8 +43,31 @@ public RawData execute(
     ) throws ProcessException {
 
         try {
-            httpNotifier.notify(callbackUrl, getWpsUrl(), getId(), callbackParams);
-            return notifiableData;
+            IOException lastException;
```
Comment (jkburges) [inline src/extension/notifier/src/main/java/au/org/emii/wps/NotifierProcess.java]: See https://github.com/aodn/geoserver-build/pull/37#discussion_r42329152

**Comment by danfruehauf** [comment]
Perhaps careful with allowing WPS to describe those parameters, they must have upper limits. The reasons are obvious I believe.

File: src/extension/notifier/src/main/java/au/org/emii/wps/NotifierProcess.java (hunk 1 above)
Comment (dnahodil) [inline src/extension/notifier/src/main/java/au/org/emii/wps/NotifierProcess.java]: I'll have a look and see how that might look.

**Comment by dnahodil** [comment]
Yeah an upper limit on retry count and and an upper and lower limit on retry interval would be required.

**Comment by dnahodil** [comment]
New PR: https://github.com/aodn/geoserver-build/pull/44

**Pull request closed without merging**